File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-27.java
class Solution {
    public int evalRPN(String[] tokens) {

        Stack<Integer> stack = new Stack<>();
        String operators[] = {"+", "-", "*", "/"};

        for(String token : tokens) {
            if(token.equals(operators[0])) {
                stack.push(stack.pop() + stack.pop());
            }
            else if(token.equals(operators[1])) {
                int a = stack.pop();
                int b = stack.pop();
                stack.push(b - a);
            }
            else if(token.equals(operators[2])) {
                stack.push(stack.pop() * stack.pop());
            }
            else if(token.equals(operators[3])) {
                int a = stack.pop();
                int b = stack.pop();
                stack.push(b / a);
            }
            else {
                stack.push(Integer.parseInt(token));
            }
        } 
        
        return stack.peek();
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-27.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = list()
        closeToOpen = {")" : "(", "]" : "[", "}" : "{"}

        #while "()" in s or "{}" in s or "[]" in s:
            #s = s.replace("()", "")
            #s = s.replace("{}", "")
            #s = s.replace("[]", "")
        #return s == ""    

        for char in s:
            if char in closeToOpen.values():
                stack.append(char)
            elif char in closeToOpen:    
                if stack and stack[-1] in closeToOpen[char]:
                    stack.pop()
                else:
                    return False        
        return not stack  
            


        